consensus/tendermint/core/upon_condition_test.go

```diff
@@ -570,6 +570,131 @@ func TestOldProposal(t *testing.T) {
+		// timely manner thus there are no old prevote yet to arrive
+		clientLockedRound := int64(rand.Intn(int(proposalValidRound)) - 1)
+
+		// the new round proposal
+		proposalMsg, proposal := generateBlockProposal(t, currentRound, currentHeight, proposalValidRound, members[currentRound].Address, false)
+
+		// old proposal some random block
+		clientLockedValue := generateBlock(currentHeight)
+
+		// the old round prevote msg to be handled to get the full quorum prevote on old round vr with value v.
+		prevoteMsg, _, _ := prepareVote(t, msgPrevote, proposalValidRound, currentHeight, proposal.ProposalBlock.Hash(), clientAddr, privateKeys[clientAddr])
+
+		// the expected prevote msg to be broadcast for the new round with <currentHeight, currentRound, proposal.ProposalBlock.Hash()>
+		prevoteMsgToBroadcast, prevoteMsgRLPNoSig, prevoteMsgRLPWithSig := prepareVote(t, msgPrevote, currentRound, currentHeight, proposal.ProposalBlock.Hash(), clientAddr, privateKeys[clientAddr])
+
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		backendMock := NewMockBackend(ctrl)
+		backendMock.EXPECT().Address().Return(clientAddr)
+
+		c := New(backendMock, config.DefaultConfig())
+		c.setCommitteeSet(committeeSet)
+		// construct round state with: old round's quorum-1 prevote for v on valid round.
+		c.messages.getOrCreate(proposalValidRound).AddPrevote(proposal.ProposalBlock.Hash(), Message{Code: msgPrevote, power: c.committeeSet().Quorum() - 1})
+
+		// client on new round's step propose.
+		c.setHeight(currentHeight)
+		c.setRound(currentRound)
+		c.setStep(propose)
+		c.lockedRound = clientLockedRound
+		c.validRound = clientLockedRound
+		c.lockedValue = clientLockedValue
+		c.validValue = clientLockedValue
+
+		//schedule the proposer timeout since the client is not the proposer for this round
+		c.proposeTimeout.scheduleTimeout(1*time.Second, c.Round(), c.Height(), c.onTimeoutPropose)
+
+		backendMock.EXPECT().VerifyProposal(*proposal.ProposalBlock).Return(time.Duration(1), nil)
+		backendMock.EXPECT().Sign(prevoteMsgRLPNoSig).Return(prevoteMsgToBroadcast.Signature, nil)
+		backendMock.EXPECT().Broadcast(context.Background(), committeeSet.Committee(), prevoteMsgRLPWithSig).Return(nil)
+
+		// now we handle new round's proposal with round_p > vr on value v.
+		err := c.handleCheckedMsg(context.Background(), proposalMsg, members[currentRound])
+		assert.Nil(t, err)
+
+		// check timer was stopped after receiving the proposal
+		assert.False(t, c.proposeTimeout.timerStarted())
+
+		// now we receive the last old round's prevote MSG to get quorum prevote on vr for value v.
+		// the old round's prevote is accepted into the round state which now have the line 28 condition satisfied.
+		// now to take the action of line 28 which was not align with pseudo code before.
+		sender := 0
+		err = c.handleCheckedMsg(context.Background(), prevoteMsg, members[sender])
+		assert.Nil(t, err)
```

Comment:
```suggestion
		assert.NoError(t, err)
```
You could make this change in a number of other places.

Reply:
Why is checking for no error better than checking for nil?

Reply:
When the test fails, the error is displayed differently to just a generic object, which makes the failure easier to read.

Reply:
@piersy I replaced them with the one you suggested to get an easier read once case failed.